### DISCUSSION TITLE
Reverting when deploying contract FundMe

### DISCUSSION THREAD

**Discussion by Joyosmit** [Cyfrin/foundry-full-course-cu#2250] [General]
This is my FundMe.sol
```
solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;
contract FundMe{
    function fund() public payable {
        require(msg.value >= 1e18, "send some more eth");
    }
}
```

When i deploy this in Remix VM, i get the following error if i send more than 0 ether:
![error_sol](https://github.com/user-attachments/assets/8a398179-467d-4601-8212-5f8ac2104100)

**Comment by EngrPips** [comment]
That is because your contract does not have a payable constructor. If you will like to send some ether to your contract on deployment then add the below code to your contract code 

```js
constructor() payable {}
```

**Reply by Joyosmit** [reply]
Thanks a lot. 
This solved the issue.

**Reply by EngrPips** [reply]
You are welcome, Keep crushing it!
